### PR TITLE
Add CI job for centos7

### LINKED ISSUE #451: Get Spicy to build on CentOS 7

### PR DESCRIPTION
**Pull request by bbannier**
This closes #451.

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -379,6 +332,45 @@ docker_alpine_3_12_task:
     path: build/spicy-*.tar.gz
     type: application/gzip
 
+docker_centos_7_task:
```
Comment (bbannier) [inline .cirrus.yml:335]: We should add the artifacts of this job to the docs.

**Comment by bbannier** [comment]
FTR, if we do not link libstdc++ statically we run into issues with number formatting, even inside a devtoolset environment.

[log.txt](https://github.com/zeek/spicy/files/5492523/log.txt)

**Comment by rsmmr** [comment]
> FTR, if we do not link libstdc++ statically we run into issues with number formatting, even inside a devtoolset environment.
> 
> [log.txt](https://github.com/zeek/spicy/files/5492523/log.txt)

Looks like hex floats are part of C99, I'm wondering if we could use libc for formatting them instead of libc++.

(I don't really like the hex floats to begin with, but they do have better precision semantics; we could also (in addition) add some logic to use the hex syntax only if we'd loose precision otherwise).

**Comment by bbannier** [comment]
> > FTR, if we do not link libstdc++ statically we run into issues with number formatting, even inside a devtoolset environment.
> > [log.txt](https://github.com/zeek/spicy/files/5492523/log.txt)
> 
> Looks like hex floats are part of C99, I'm wondering if we could use libc for formatting them instead of libc++.
> 
> (I don't really like the hex floats to begin with, but they do have better precision semantics; we could also (in addition) add some logic to use the hex syntax only if we'd loose precision otherwise).

I tested this and hexfloat support via libc seems to work even on centos7. Added a patch to that effect.

File: hilti/toolchain/src/compiler/codegen/ctors.cc [hunk 2]
```diff
@@ -92,8 +94,10 @@ struct Visitor : hilti::visitor::PreOrder<std::string, Visitor> {
     result_t operator()(const ctor::Port& n) { return fmt("hilti::rt::Port(\"%s\")", n.value()); }
 
     result_t operator()(const ctor::Real& n) {
-        // hexfloat format for lossless output, at most 13 fraction hexits
-        return fmt("%a", n.value());
+        constexpr size_t size = 256;
```
Comment (rsmmr) [inline hilti/toolchain/src/compiler/codegen/ctors.cc:97]: Worth a comment why we aren't using `fmt()` here.

File: hilti/toolchain/src/compiler/visitors/printer.cc [hunk 3]
```diff
@@ -236,7 +238,12 @@ struct Visitor : visitor::PreOrder<void, Visitor> {
 
     void operator()(const ctor::Port& n) { out << n.value(); }
 
-    void operator()(const ctor::Real& n) { out << fmt("%a", n.value()); }
+    void operator()(const ctor::Real& n) {
```
Comment (rsmmr) [inline hilti/toolchain/src/compiler/visitors/printer.cc:241]: same

**Review by rsmmr — APPROVED** [review]
Cool, thanks!

**Pull request merged**